packages/block-editor/src/components/global-styles/color-panel.js

```diff
@@ -0,0 +1,754 @@
+				{
+					key: 'link',
+					label: __( 'Default' ),
+					inheritedValue: linkColor,
+					setValue: setLinkColor,
+					userValue: userLinkColor,
+				},
+				{
+					key: 'hover',
+					label: __( 'Hover' ),
+					inheritedValue: hoverLinkColor,
+					setValue: setHoverLinkColor,
+					userValue: userHoverLinkColor,
+				},
+			],
+		},
+	].filter( Boolean );
+
+	elements.forEach( ( { name, label, showPanel } ) => {
+		if ( ! showPanel ) return;
+
+		const elementBackgroundColor = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.background
+		);
+		const elementGradient = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.gradient
+		);
+		const elementTextColor = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.text
+		);
+		const elementBackgroundUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.background
+		);
+		const elementGradientUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.gradient
+		);
+		const elementTextUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.text
+		);
+		const hasElement = () =>
+			!! (
+				elementTextUserColor ||
+				elementBackgroundUserColor ||
+				elementGradientUserColor
+			);
+		const resetElement = () => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							background: undefined,
+							gradient: undefined,
+							text: undefined,
+						},
+					},
+				},
+			} );
+		};
+
+		const setElementTextColor = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							text: encodeColorValue( newValue ),
+						},
+					},
+				},
+			} );
+		};
+		const setElementBackgroundColor = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							background: encodeColorValue( newValue ),
+							gradient: undefined,
+						},
+					},
+				},
+			} );
+		};
+		const setElementGradient = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							gradient: encodeGradientValue( newValue ),
+							background: undefined,
+						},
+					},
+				},
+			} );
+		};
+		const supportsTextColor = true;
+		const supportsBackground = name !== 'caption';
+
+		items.push( {
+			key: name,
+			label,
+			hasValue: hasElement,
+			resetValue: resetElement,
+			isShownByDefault: defaultControls[ name ],
+			indicators:
+				supportsTextColor && supportsBackground
+					? [
+							elementTextColor,
+							elementBackgroundColor ?? elementGradient,
```

Comment:
Should this line be the other way around? This preferences background color to be displayed before gradient color. So, in TwentyTwentyThree theme, which has a background color of white set in its `theme.json` file, if I set a gradient color, then it gets displayed here as white:

<img width="270" src="https://user-images.githubusercontent.com/14988353/227818397-ab6034bc-6a3b-4b10-adbb-e1920184fb05.png" />

If I remove the background color from the `theme.json` file, then it displays the gradient as expected:

<img width="270" alt="image" src="https://user-images.githubusercontent.com/14988353/227818459-4565e060-353b-4021-a325-018af1b4c577.png">